### PR TITLE
feat: basic support Apache APISIX 2.9

### PR DESCRIPTION
**Pull request by bzp2010**
**Why submit this pull request?**

- [ ] Bugfix
- [ ] New feature provided
- [x] Improve performance
- [ ] Backport patches

**What changes will this PR take into?**

Provide basic support for Apache APISIX 2.9. Includes changes to the upstream property definition.

**Checklist:**

- [x] Did you explain what problem does this PR solve? Or what new features have been added?
- [x] Have you added corresponding test cases?
- [x] Have you modified the corresponding document?
- [x] Is this PR backward compatible? If it is not backward compatible, please discuss on the mailing list first

### CODE REVIEW

**Comment by liuxiran** [comment]
@nic-chen please help to review, thanks a lot

File: .github/workflows/backend-cli-test.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        etcd: [3.4.13]
+        etcd: [3.4.14]
```
Comment (nic-chen) [inline .github/workflows/backend-cli-test.yml:20]: we could update to 3.4.16

File: .github/workflows/go-lint.yml [hunk 2]
```diff
@@ -12,44 +12,8 @@ on:
...
-      - uses: actions/checkout@v2
-
-      - name: golangci-lint
-        uses: golangci/golangci-lint-action@v2
-        with:
-          version: latest
-          working-directory: api
-          args: --tests=false
-          only-new-issues: true
-
```
Comment (nic-chen) [inline .github/workflows/go-lint.yml:48]: why remove these?

File: api/conf/schema.json [hunk 3]
```diff
@@ -2601,37 +2720,10 @@
...
-					"service_version": {
-						"pattern": "^\\d+\\.\\d+\\.\\d+",
-						"type": "string"
-					}
-				},
-				"required": ["service_name", "service_version"],
-				"type": "object"
-			},
-			"version": 0.1
-		},
```
Comment (nic-chen) [inline api/conf/schema.json:2629]: we should keep it.

File: api/conf/schema.json [hunk 4]
```diff
@@ -2666,87 +2758,8 @@
...
-				"$comment": "this is a mark for our injected plugin schema",
-				"properties": {
-					"disable": {
-						"type": "boolean"
-					}
-				},
-				"type": "object"
-			},
-			"version": 0.1
-		},
```
Comment (nic-chen) [inline api/conf/schema.json:2746]: ditto

File: api/conf/schema.json [hunk 5]
```diff
@@ -3526,27 +3625,17 @@
...
-				"additionalProperties": false,
-				"properties": {
-					"disable": {
-						"type": "boolean"
-					}
-				},
-				"type": "object"
-			},
-			"version": 0.1
-		},
```
Comment (nic-chen) [inline api/conf/schema.json:3549]: ditto

File: api/conf/schema.json [hunk 6]
```diff
@@ -3586,20 +3675,6 @@
...
```
Comment (nic-chen) [inline api/conf/schema.json:3602]: ditto

File: api/conf/schema.json [hunk 7]
```diff
@@ -4076,49 +4192,6 @@
...
-						"default": 1,
-						"maximum": 1,
-						"minimum": 0.00001,
-						"type": "number"
-					}
-				},
-				"type": "object"
-			},
-			"version": 0.1
-		},
```
Comment (nic-chen) [inline api/conf/schema.json:4121]: ditto

File: api/test/e2enew/upstream/upstream_retry.go [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+	ginkgo.It("delete upstream", func() {
+		base.RunTestCase(base.HttpTestCase{
+			Object:       base.ManagerApiExpect(),
+			Method:       http.MethodDelete,
+			Path:         "/apisix/admin/upstreams/retry",
+			Headers:      map[string]string{"Authorization": base.GetToken()},
+			ExpectStatus: http.StatusOK,
+		})
+	})
+})
```
Comment (nic-chen) [inline api/test/e2enew/upstream/upstream_retry.go:63]: Need dp test, please refer to APISIX related test

File: api/internal/core/entity/entity.go [hunk 9]
```diff
@@ -133,16 +134,16 @@ type UnHealthy struct {
...
-	Concurrency            int       `json:"concurrency,omitempty"`
-	Host                   string    `json:"host,omitempty"`
-	Port                   int       `json:"port,omitempty"`
-	HTTPPath               string    `json:"http_path,omitempty"`
-	HTTPSVerifyCertificate string    `json:"https_verify_certificate,omitempty"`
-	Healthy                Healthy   `json:"healthy,omitempty"`
-	UnHealthy              UnHealthy `json:"unhealthy,omitempty"`
-	ReqHeaders             []string  `json:"req_headers,omitempty"`
+	Type                   string       `json:"type,omitempty"`
+	Timeout                TimeoutValue `json:"timeout,omitempty"`
```
Comment (starsz) [inline api/internal/core/entity/entity.go:138]: Why change this to float32?

File: .github/workflows/go-lint.yml (hunk 2 above)
Comment (liuxiran) [reply .github/workflows/go-lint.yml:48]: keep golangci

File: api/internal/core/entity/entity.go [hunk 10]
```diff
@@ -97,10 +97,11 @@ type Route struct {
 }
 
 // --- structures for upstream start  ---
+type TimeoutValue float32
```
Comment (liuxiran) [inline api/internal/core/entity/entity.go:100]: why

File: api/conf/schema.json [hunk 11]
```diff
@@ -4903,4 +5030,4 @@
 			"version": 0.1
 		}
 	}
-}
+}
```
Comment (liuxiran) [inline api/conf/schema.json:5033]: line

File: api/internal/core/entity/entity.go (hunk 10 above)
Comment (bzp2010) [reply api/internal/core/entity/entity.go:100]: There are two "number" definitions in jsonschema, one is an `integer`, the other is `number`, where `number` runs to store float values. Such as 5.5

Therefore, a float value is used here to store.

In addition, only map[string]interface{} was used to store the value before, and there was ambiguity in the data type.  Crash may occur during runtime.  This is also the reason for this modification.（add a Timeout struct）

File: api/internal/core/entity/entity.go (hunk 9 above)
Comment (bzp2010) [reply api/internal/core/entity/entity.go:138]: There are two "number" definitions in jsonschema, one is an integer, the other is number, where number runs to store float values. Such as 5.5

Therefore, a float value is used here to store.

In addition, only map[string]interface{} was used to store the value before, and there was ambiguity in the data type. Crash may occur during runtime. This is also the reason for this modification.（add a Timeout struct）

File: .github/workflows/backend-cli-test.yml (hunk 1 above)
Comment (liuxiran) [reply .github/workflows/backend-cli-test.yml]: there is no image of etcd:3.4.16, so just keep 3.4.14 here

File: api/internal/core/entity/entity.go (hunk 10 above)
Comment (liuxiran) [reply api/internal/core/entity/entity.go:100]: cc @nic-chen @starsz

Comment (liuxiran) [reply api/internal/core/entity/entity.go:100]: so do we support timeout 5.5s?

Comment (liuxiran) [reply api/internal/core/entity/entity.go:100]: image

It seems that `float` is ok

**Comment by liuxiran** [comment]
all CI passed @starsz @nic-chen please review again, thanks a lot

**Review by nic-chen — APPROVED** [review]

**Review by starsz — APPROVED** [review]

**Review by imjoey — APPROVED** [review]

**Pull request merged**